script.js

```diff
@@ -1,4 +1,420 @@
+    }
+    index = index + 1;
+    while (index < aceCounter) {
+      totalHandValue = totalHandValue - 10;
+    }
+    index = index + 1;
+  }
+
+  return totalHandValue;
+};
+
+// ==== Function for shuffled deck ====
+var shuffledDeck = function () {
+  var newDeck = makeDeck();
+  var shuffledDeck = shuffleCards(newDeck);
+  return shuffledDeck;
+};
+
+// ==== function to reset game ====
+var reset = function () {
+  currentMode = gameStart;
+  gameDeck = "empty at the start";
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // 1st click - click to shuffle cards
+  var message = "";
+  if (currentGameMode == gameStart) {
+    gameDeck = shuffledDeck();
+
+    playerHand.push(gameDeck.pop());
+    playerHand.push(gameDeck.pop());
+    computerHand.push(gameDeck.pop());
+    computerHand.push(gameDeck.pop());
+
+    currentGameMode = drawCards;
+    // Approach through little steps first e.g. shuffle first etc
+    message = "Ya'll been dealt cards bruh";
+
+    return message;
+  }
+  // 2nd click - check for blackjack
+  if (currentGameMode == drawCards) {
+    var playerBlackjack = BlackjackCheck(playerHand);
+    var comBlackJack = BlackjackCheck(computerHand);
+
+    //  player & com have blackjack = draw
+    //  boolean false add ! in front of var
+    if (playerBlackjack || comBlackJack) {
+      if (playerBlackjack && comBlackJack) {
+        message =
+          showPlayerAndDealerHands(playerHand, computerHand) +
+          "<br>" +
+          "It's a blackjack tie mannn! " +
+          "<br>" +
+          showHandTotalValues(playerHandTotalValue, compHandTotalValue);
+      }
+      // player has blackjack = player wins
+      else if (playerBlackjack && !comBlackJack) {
+        message =
+          showPlayerAndDealerHands(playerHand, computerHand) +
+          "<br>" +
+          "Eyyyy! You won mannnn! " +
+          "<br>" +
+          showHandTotalValues(playerHandTotalValue, compHandTotalValue);
+      }
+      // com has blackjack = com wins
+      else {
+        message =
+          showPlayerAndDealerHands(playerHand, computerHand) +
+          "<br>" +
+          "Nawwww! Dealer won mannn! " +
+          "<br>" +
+          showHandTotalValues(playerHandTotalValue, compHandTotalValue);
+      }
+    } else {
+      message =
+        showPlayerAndDealerHands(playerHand, computerHand) +
+        "<br>" +
+        "Nope! There is no Blackjack. " +
+        "<br>" +
+        showHandTotalValues(playerHandTotalValue, compHandTotalValue);
+      // no blackjack - game continues >>
+      currentGameMode = gameResults;
+      console.log(message);
+    }
+
+    //  calculate sum of hand value & com
+    var playerHandTotalValue = totalHandValue(playerHand);
+    var compHandTotalValue = totalHandValue(computerHand);
+
+    //  compare total hand valuer
+    //   same value = draw
+    if (playerHandTotalValue == compHandTotalValue) {
```

Comment:
Same here. This section is basically comparing the hand values and getting a message. Refactor it into a helper function